File: Week_08/G20200343030001/Leetcode_557_001.java
package Week_08;

public class Leetcode_557_001 {
    public String reverseWords(String s) {
        StringBuilder sb = new StringBuilder();
        String[] words = s.split(" ");

        for (String word: words) {
            sb.append(new StringBuilder(word).reverse());
            sb.append(" ");
        }

        return sb.toString().trim();
    }
}

File: Week_08/G20200343030005/Leetcode_557_001.js
// 557. 反转字符串中的单词 III https://leetcode-cn.com/problems/reverse-words-in-a-string-iii/

/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function (s) {
  let arr = s.split("").reverse().join("");
  return arr.split(" ").reverse().join(" ");
};

var reverseWords = function (s) {
  return s.split("").reverse().join("").split(/\s+/).reverse().join(" ");
};
